src/main/java/rip/bolt/ingame/ranked/RankedManager.java

```diff
@@ -109,17 +112,43 @@ private void updateMatch(BoltMatch match) {
       return;
     }
 
-    List<MatchPlayer> players =
+    MatchManager mm = PGM.get().getMatchManager();
+
+    List<UpdateRecord> updates =
         match.getTeams().stream()
             .map(Team::getParticipations)
             .flatMap(Collection::stream)
             .map(Participation::getUser)
-            .map(rankManager::notifyUpdates)
-            .filter(Objects::nonNull)
+            .map(
+                u ->
+                    new UpdateRecord(u, this.match.getUser(u.getUuid()), mm.getPlayer(u.getUuid())))
+            .filter(UpdateRecord::isValid)
             .collect(Collectors.toList());
 
     this.match = match;
-    players.forEach(rankManager::updatePlayer);
+    if (updates.isEmpty()) return;
+
+    Match pgmMatch = updates.get(0).player.getMatch();
+    pgmMatch.callEvent(new MatchStatsEvent(pgmMatch, true, true));
```

Comment:
Use `Bukkit.getPluginManager().callEvent(...` and inline the pgmMatch argument which removes the not very nice `pgmMatch` var.